apps/vor/pkg/school/school.go

```diff
@@ -83,7 +86,87 @@ func getClasses(server rest.Server, store postgres.SchoolStore) http.Handler {
 		return nil
 	})
 }
+func getClassSession(server rest.Server, store postgres.SchoolStore) http.Handler {
+	type responseBody struct {
+		Date string `json:"date"`
+	}
+	return server.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+		classId := chi.URLParam(r, "classId")
+		attendance, err := store.GetClassSession(classId)
+		var response []responseBody
+
+		if len(attendance) > 0 {
+			attend := make(map[string]int)
+			for _, attendance := range attendance {
+				attend[attendance.StudentId] = 1
+				response = append(response, responseBody{
+					Date: attendance.Date.String(),
+				})
+			}
+		}
+
+		if err != nil {
+			return &rest.Error{
+				Code:    http.StatusInternalServerError,
+				Message: "failed querying attendance",
+				Error:   err,
+			}
+		}
+		if err := rest.WriteJson(w, response); err != nil {
+			return rest.NewWriteJsonError(err)
+		}
+
+		return nil
+	})
+}
+func getClassAttendance(server rest.Server, store postgres.SchoolStore) http.Handler {
+	type responseBody struct {
+		StudentId string `json:"studentId"`
+		Name      string `json:"name"`
+		Absent    bool   `json:"absent"`
+	}
+	return server.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+		session := chi.URLParam(r, "session")
+		classId := chi.URLParam(r, "classId")
+		attendance, err := store.GetClassAttendance(classId, session)
+		var response []responseBody
+
+		if len(attendance) > 0 {
+			students := attendance[0].Class.Students
+			attend := make(map[string]int)
+			for _, attendance := range attendance {
+				attend[attendance.StudentId] = 1
+				response = append(response, responseBody{
+					StudentId: attendance.StudentId,
+					Name:      attendance.Student.Name,
+					Absent:    true,
```

Comment:
I feel like this might be flipped. Isn't this loop is for students who exist on the attendance table, therefore not absent during the class session?

Reply:
The attendance table, only include those that are present. So we need a way to filter students from that class. That variables is to hold which student are present at the class for that session. And then we loop through the list of students of that class and check if the studentId is not in the 'attend' variables, it means the student is absent

Reply:
So this loop checks for students who are present, and the next one below checks who are absent?

Reply:
yes

Reply:
Oh ya. This one should be Absent :false, and the latter Absent :true